subsys/net/lib/mqtt_helper/mqtt_helper.c

```diff
@@ -150,40 +153,79 @@ static int certificates_provision(void)
 		return 0;
 	}
 
-	err = tls_credential_add(CONFIG_AZURE_IOT_HUB_SEC_TAG,
-				 TLS_CREDENTIAL_CA_CERTIFICATE,
-				 ca_certificate,
-				 sizeof(ca_certificate));
-	if (err < 0) {
-		LOG_ERR("Failed to register CA certificate: %d",
-			err);
-		return err;
+	if (sizeof(ca_certificate) > 0) {
+		err = tls_credential_add(CONFIG_MQTT_HELPER_SEC_TAG,
+					 TLS_CREDENTIAL_CA_CERTIFICATE,
+					 ca_certificate,
+					 sizeof(ca_certificate));
```

Comment:
I recall I had issues when sending in null-terminated certificates at some point, but this is perhaps not an issue anymore?

Reply:
I wasn't aware. Has been working fine but I guess its safer to avoid sending in the null null terminated certificate.

Reply:
Alright, then my recollection is probably wrong on this